### PR TITLE
DT-27: add subnet and firewall changes for CPM

### PR DESCRIPTION
**Pull request by hugh-emerson**

### CODE REVIEW

File: terraform/modules/networking/subnet.tf [hunk 1]
```diff
@@ -105,11 +106,21 @@ resource "aws_subnet" "github_runner" {
...
+  count                   = 3
+  cidr_block              = cidrsubnet(local.cpm_private_cidr_10, 2, count.index)
+  vpc_id                  = aws_vpc.vpc.id
+  availability_zone       = data.aws_availability_zones.available.names[count.index]
+  map_public_ip_on_launch = false
+  tags                    = { Name = "cpm-private-subnet-${data.aws_availability_zones.available.names[count.index]}-${var.environment}" }
+}
+
+resource "aws_subnet" "vpc_endpoints_subnets" {
+  count             = 3
```
Comment (hugh-emerson) [inline terraform/modules/networking/subnet.tf:119]: @BenRamchandani I know you said you don't think a subnet per AZ is actually necessary, but the docs recommended it and I don't see the downside?

Comment (BenRamchandani) [reply terraform/modules/networking/subnet.tf:119]: Agree we should do it, but assuming I'm reading this correctly it does cost money, <https://aws.amazon.com/privatelink/pricing/>.

```text
0.011$/hour * 720 hours/month * 2 AZs * 12 endpoints = 190.08$/month
```

So about £170/month for each environment. Fine for production and staging should match, not sure whether it's worth the extra complexity of going down to one subnet to make test a bit cheaper.

File: manual_scripts/active_directory/set-password-policy.ps1 [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Policy taken from Datamart prod
+Set-ADFineGrainedPasswordPolicy -Identity CustomerPSO-03 `
```
Comment (BenRamchandani) [inline manual_scripts/active_directory/set-password-policy.ps1:2]: Nitpick should we give it a different name, I assume `CustomerPSO-03` doesn't mean anything?

File: terraform/production/outputs.tf [hunk 3]
```diff
@@ -10,6 +10,10 @@ output "public_subnet_ids" {
   value = module.networking.public_subnets[*].id
 }
 
+output "cpm_private_subnet_ids" {
```
Comment (BenRamchandani) [inline terraform/production/outputs.tf:13]: Do we need these outputs?

File: manual_scripts/active_directory/set-password-policy.ps1 (hunk 2 above)
Comment (hugh-emerson) [reply manual_scripts/active_directory/set-password-policy.ps1:2]: These are created by AWS https://docs.aws.amazon.com/directoryservice/latest/admin-guide/supportedpolicysettings.html

File: terraform/production/outputs.tf (hunk 3 above)
Comment (hugh-emerson) [reply terraform/production/outputs.tf:13]: This is so the CPM repo can access these.

File: manual_scripts/active_directory/set-password-policy.ps1 (hunk 2 above)
Comment (BenRamchandani) [reply manual_scripts/active_directory/set-password-policy.ps1:2]: :+1: maybe worth a comment with a link then

File: terraform/modules/networking/main.tf [hunk 4]
```diff
@@ -25,7 +25,8 @@ locals {
       tls_allowed_domains = [
         "github.com", "api.github.com", "codeload.github.com",
         "objects.githubusercontent.com", "objects-origin.githubusercontent.com", "github-releases.githubusercontent.com", "github-registry-files.githubusercontent.com",
-        ".actions.githubusercontent.com"
+        ".actions.githubusercontent.com",
+        "logs.${data.aws_region.current.name}.amazonaws.com", "ec2messages.${data.aws_region.current.name}.amazonaws.com"
```
Comment (BenRamchandani) [inline terraform/modules/networking/main.tf:29]: Do we need these?

File: terraform/modules/networking/outputs.tf [hunk 5]
```diff
@@ -55,6 +55,11 @@ output "github_runner_private_subnet" {
   description = "Private /24 subnet for GitHub runner instance"
 }
 
+output "cpm_private_subnets" {
+  value       = aws_subnet.cpm_private
+  description = "Three /24 subnets for GitHub runner instance"
```
Comment (BenRamchandani) [inline terraform/modules/networking/outputs.tf:60]:
```suggestion
  description = "Three private /24 subnets CPM"
```

**Review by BenRamchandani — APPROVED** [review]
LGTM, will conflict a bit with #53

**Comment by hugh-emerson** [comment]
I'll fold https://github.com/communitiesuk/delta-common-infrastructure/pull/53 into this

**Pull request merged**